File: kattis/splat/splat.py

from math import acos

splats = []

c = int(input())
for _ in range(c):
	n = int(input())	# number of paint drops
	for i in range(n):
		
		# (x, y) = location of paint drop, v = volume
		inp = [i for i in input().split()]
		x = float(inp[0])
		y = float(inp[1])
		v = float(inp[2])
		col = inp[3]
		rad = (v / acos(-1)) ** 0.5
		splats.append((x, y, v, rad, col))

	m = int(input())
	for i in range(m):
		x, y = [float(i) for i in input().split()]

		ans = "white"
		for j in range(len(splats)-1, -1, -1):
			dx = x - splats[j][0]
			dy = y - splats[j][1]

			if dx*dx + dy*dy <= splats[j][3]*splats[j][3] + 1e-8:
				ans = splats[j][4]
				break

		print(ans)

File: kattis/splat/splat.cpp
#include <iostream>
#include <cmath>
#include <string>
#define EPS 1e-8
using namespace std;

struct Splat{
	double x, y, r;
	string color;
};

int main(){

	int c;
	cin >> c;

	Splat splats[105];
	while (c--){
		int n;
		cin >> n;

		for (int i = 0; i < n; ++i){
			double v;
			cin >> splats[i].x >> splats[i].y >> v 
				>> splats[i].color;
			splats[i].r = sqrt( v / acos(-1.0));
		}

		int m;
		cin >> m;
		while (m--){
			double x, y;
			cin >> x >> y;

			string col = "white";

			for (int i = n - 1; i >= 0; --i){
				double dx = x - splats[i].x;
				double dy = y - splats[i].y;
				if (dx*dx + dy*dy <= splats[i].r*splats[i].r + EPS){
					col = splats[i].color;
					break;
				}
			}
			cout << col << endl;

		}
	}

	return 0;
}
